bin/drop_sample_annot.py

```diff
@@ -50,24 +55,74 @@ def is_paired_end(single_end: str) -> bool:
 
 
+def is_stranded(strandedness: str) -> str:
+    """Logical funciton to determine sample strandness"""
+    if strandedness.lower() == "reverse":
+        return "reverse"
+    elif strandedness.lower() == "forward":
+        return "yes"
+    else:
+        return "no"
+
+
+def count_mode(sample_count_mode: str) -> str:
+    """Logical funciton to determine if count mode is given or default "IntersectionStrict" should be used"""
+    if isna(sample_count_mode) or sample_count_mode == "" or sample_count_mode == "NA":
+        return "IntersectionStrict"
+    else:
+        return sample_count_mode
+
+
+def count_overlaps(sample_count_overlap: str) -> str:
+    """Logical funciton to determine if count overlap is given or default "TRUE" should be used"""
+    if (
+        isna(sample_count_overlap)
+        or sample_count_overlap == ""
+        or sample_count_overlap == "NA"
+    ):
+        return True
+    else:
+        return sample_count_overlap
+
+
 def write_final_annot_to_tsv(ref_count_file: str, ref_annot: str, out_file: str):
     """
     Concatenates the Sample Annotation produced by SampleAnnotation with the one
-    provided for the reference samples, checking for duplicate sample IDs
+    provided for the reference samples, if one is provided, checking for duplicate sample IDs
     """
     df_samples: DataFrame = read_csv("drop_annotation_given_samples.tsv", sep="\t")
-    df_reference: DataFrame = read_csv(ref_annot, sep="\t")
-    df_reference["GENE_COUNTS_FILE"] = ref_count_file
-    df_reference["SPLICE_COUNTS_DIR"] = df_reference["SPLICE_COUNTS_DIR"].str.rstrip("/").apply(os.path.basename)
-    df_reference["DROP_GROUP"] = df_reference["DROP_GROUP"].str.replace(" ", "")
-    df_samples["COUNT_OVERLAPS"] = df_reference["COUNT_OVERLAPS"].iloc[0]
-    df_samples["COUNT_MODE"] = df_reference["COUNT_MODE"].iloc[0]
-    df_samples["HPO_TERMS"] = df_reference["HPO_TERMS"].iloc[0]
-    for id in df_samples["RNA_ID"]:
-        df_reference = df_reference[df_reference["RNA_ID"].str.contains(id) == False]
-    df: DataFrame = concat([df_samples, df_reference]).reset_index(drop=True)
-    df.fillna("NA", inplace=True)
-    df.to_csv(out_file, index=False, sep="\t")
+    if ref_annot == "None" or ref_count_file == "None":
+        print(
+            "No reference samples were provided by the user see usage of --ref_count_file and --ref_annot if you want to provide reference samples"
+        )
+        if df_samples.shape[0] < 50:
+            print(
+                "At least 30 samples are required for Aberrant Splicing and 50 for Aberrant expression"
+            )
+            print(f"Only {df_samples.shape[0]} samples were provided by the user")
+        df_samples.fillna("NA", inplace=True)
+        df_samples["COUNT_MODE"] = "IntersectionStrict"
+        df_samples["COUNT_OVERLAPS"] = True
+        df_samples.to_csv(out_file, index=False, sep="\t")
+    else:
+        df_reference: DataFrame = read_csv(ref_annot, sep="\t")
```

Comment:
Here you are type hinting, it is nice and you could try using it more

Reply:
Sure, I will do so on one of the next PRs